src/features/Navigation/MobileDropdown.tsx

```diff
@@ -50,35 +51,57 @@ export const MobileDropdown: React.FC<Props> = ({ items }) => {
           ))}
 
           <Divider />
-
-          {infoItems.map(item => (
-            <UnstyledLink
-              key={item.text}
-              href={item.url}
-              target="_blank"
-              rel="noreferrer"
-            >
-              <Text variant="linkMobile" color="purple">
-                {item.text}
-              </Text>
-            </UnstyledLink>
-          ))}
+          <UnstyledLink
+            href={t(`navigation.info.link.feedback.url`)}
+            target="_blank"
+            rel="noreferrer"
+          >
+            <Text variant="linkMobile" color="purple">
+              {t('navigation.info.link.feedback.text')}
+            </Text>
+          </UnstyledLink>
+          <UnstyledLink
+            href={t(`navigation.info.link.termsAndPrivacy.url`)}
+            target="_blank"
+            rel="noreferrer"
+          >
+            <Text variant="linkMobile" color="purple">
+              {t('navigation.info.link.termsAndPrivacy.text')}
+            </Text>
+          </UnstyledLink>
 
           <Divider />
-
-          {languages.map(lang => (
-            <UnstyledDiv key={lang.code} onClick={handleLangChange}>
-              <Text variant="linkMobile" color="purple">
-                {lang.label}
-              </Text>
-            </UnstyledDiv>
-          ))}
+          <UnstyledDiv
+            isSelected={isSelected('en')}
+            onClick={e =>
+              isSelected('en') ? e.stopPropagation() : changeLanguage('en')
+            }
+          >
+            <Text
+              color={isSelected('en') ? 'darkblue' : 'purple'}
+              variant={isSelected('en') ? 'linkDisabled' : 'linkMobile'}
+            >
+              {t(`navigation.language.en.long`)}
+            </Text>
+          </UnstyledDiv>
+          <UnstyledDiv
+            isSelected={isSelected('fi')}
+            onClick={e =>
+              isSelected('fi') ? e.stopPropagation() : changeLanguage('fi')
+            }
+          >
+            <Text
+              color={isSelected('fi') ? 'darkblue' : 'purple'}
+              variant={isSelected('fi') ? 'linkDisabled' : 'linkMobile'}
+            >
+              {t(`navigation.language.fi.long`)}
+            </Text>
+          </UnstyledDiv>
```

Comment:
For this we define first a new component:

![image](https://user-images.githubusercontent.com/34128180/210266712-7d721d1f-9e48-4d53-b297-f36dfcfee4e3.png)


Then we can use it like this:

![image](https://user-images.githubusercontent.com/34128180/210266776-1b1d2e45-d585-4ad8-b549-037ae4f4a84c.png)

Reply:
The other way works too, but there is too much ternary operations and it's a bit messy looking. This way we can abstract the logic to only one place :p

Reply:
I would define this component in a new file, maybe call it MobileLanguageItem,
I would move the UnstyledDiv-there too (and rename it, because its not a DIV but an A.. 
It's my code though, but thanks already for fixing it 8)=D!

Reply:
We could also define the type Lang = 'fi' | 'en', and use it instead of typing languagecode as string 
In my example just use the Lang instead of the union type

Reply:
![image](https://user-images.githubusercontent.com/34128180/210268036-b605960a-de6d-41db-83fd-385099cc7727.png)
 Cool thing is that the ts-cries if we had extra-lang there 'swe' for the t-function, because its typed correctly =)

Reply:
Fixed! :)